user/languages/_includes/modern-c-cpp-toolchain.md

```diff
@@ -0,0 +1,130 @@
+### Modern C/C++ and Toolchain Versioning
+
+If you need a more modern toolchain or more granular control over your tools, you will have to handle that yourself. Fortunately, the community has already whitelisted the [repositories](https://github.com/travis-ci/apt-source-whitelist/blob/master/ubuntu.json) and [packages](https://github.com/travis-ci/apt-package-whitelist/blob/master/ubuntu-precise) necessary for most such upgrades.
```

Comment:
That was actually intentional, and it remains true today: clang-3.9 is not available, and you also can't version cmake (the latest `cmake` that you can upgrade to is 3.2.3, which was [released in June 2015](https://blog.kitware.com/cmake-3-2-3-available-for-download/)).

Perhaps instead a note about how to request that packages be be whitelisted?